openprescribing/matrixstore/sql_functions.py

```diff
@@ -13,8 +15,13 @@ def step(self, value):
             return
         matrix = deserialize(value)
         if self.accumulator is None:
-            self.accumulator = zeros_like(matrix)
-        self.accumulator += matrix
+            # We need this to be in Fortran (i.e. column-major) order for the
+            # fast addition path below to work
+            self.accumulator = zeros_like(matrix, order='F')
+        try:
+            fast_in_place_add(self.accumulator, matrix)
+        except FastInPlaceAddError:
```

Comment:
Would we expect to hit this?

Reply:
Yes, but only rarely. Where a matrix is sufficiently populated (I think 50% is our current cutoff) we store it as a dense matrix and so this technique wouldn't apply. This is the case with the practice statistics matrices and possibly also with some very commonly prescribed presentations although I haven't checked the precise numbers on those.